### ISSUE TITLE
UI locks up when IINA left running in the background

### ISSUE THREAD

**Issue by svobs** [iina/iina#4669]
**System and IINA version:**

- macOS 14.0 Sonoma
- IINA 1.3.3

**Expected behavior:**
IINA window should sleep quietly and then be able to resume playback when brought to the front.

**Actual behavior:**

Window stopped responding and looks almost like it's not the active ("main") window.

Screen recording below. Before the video I tried using my key binding for opening the playlist panel and got beeped at. But then I was able to open the playlist from the menu before it became grayed out, and I cannot close it by any method. Notice also that the window's traffic light buttons look inactive. But I can sometimes get them to light up for a little while by changing apps back and forth.

https://github.com/iina/iina/assets/2213815/1a5d38aa-45cc-446c-8543-d30c64461164

<br/>
The video playing is this simple test video I made using ffmpeg:

https://github.com/iina/iina/assets/2213815/3620ddcd-2c10-48b7-9dbf-648e1e2e66e2

Logs and process sample:
[2023-10-23-IINA-Hang.zip](https://github.com/iina/iina/files/13111447/2023-10-23-IINA-Hang.zip)

The process is using 0% CPU now, 15 threads. 409 MB RAM.

**Steps to reproduce:**

Dragged & dropped a test video onto IINA, let it run until the end and then went and used other apps for a while (not long - maybe 10 min). Not sure if I navigated away before or after the video ended.

- [x] MPV does not have this problem.

**How often does this happen?**

Not sure. I may have seen it in test builds and dismissed it as my fault, but this is the last official release.

**Comment by svobs** [comment]
And...it's back. From outer space. Like nothing happened.

Updated logs attached below. Funny that the first thing it reports is a `mouseUp`. I made a few test clicks and then hit the play button to play it back all the way through.
[2023-10-23-22-33-24_ELMLc1.zip](https://github.com/iina/iina/files/13111854/2023-10-23-22-33-24_ELMLc1.zip)

**Comment by low-batt** [comment]
I was expecting to see some sort of lockup in the process sample, but it looks like everything is idle. Very odd.

The IINA log shows a mpv config file is being used that triggers errors and warnings from mpv such as:
```text
22:33:24.964 [mpv][e] [cplayer] error: /Users/msvoboda/.config/mpv/mpv.conf:51: setting option gpu-api='vulkan' failed.
```

I also see use of IINA plugins.

Not sure if either of those are responsible for this behavior, but it certainly would be good to eliminate the errors and warnings from mpv.

I agree, the dim window buttons indicate trouble with focus. If you search the net you will find instances of such problems that were traced to 3<sup>rd</sup> party applications. A reply to the question [Is there a way to detect what program is stealing focus on my Mac?](https://apple.stackexchange.com/questions/123730/is-there-a-way-to-detect-what-program-is-stealing-focus-on-my-mac) contains a python script that reports which app has the focus. Maybe some other app is having trouble with macOS Sonoma and grabbing the focus?

**Comment by svobs** [comment]
In my experience, a lockup of this type is the hallmark of a network-related issue. Very unlikely to be anything else. Either IINA was blocked waiting for a request with no timeout, or it was blocking until it got some desired state change from a remote server. It reminds me of [this infamous event](https://arstechnica.com/gadgets/2020/11/macos-big-sur-launch-appears-to-cause-temporary-slowdown-in-even-non-big-sur-macs/).

I haven't spent any time in these corners of IINA but I'll go through the list of suspects:
1. IINA plugins. I have 2 test plugins installed, but neither was enabled at the time. Possibly the plugin infrastructure was making connections to GitHub. But GitHub is known for being one of the highest-availability-rate sites on the Internet. Seems less likely.
2. youtube-dl: I did have this enabled during the test, and I do have it installed. I've never actually used it though. Can't rule it out.
3. opensubtitles[.org|.com]: seems like a prime suspect, especially given its network issues yesterday. I will take a look at this code.
4. Other Lua scripts: I have only `mpv-splice` installed. Will examine it as well.
5. Check for updates: disabled for me.
6. Something else? As far as I know, IINA doesn't make any other network connections, but would love to learn if it does.

**Comment by svobs** [comment]
> I agree, the dim window buttons indicate trouble with focus. If you search the net you will find instances of such problems that were traced to 3rd party applications. A reply to the question [Is there a way to detect what program is stealing focus on my Mac?](https://apple.stackexchange.com/questions/123730/is-there-a-way-to-detect-what-program-is-stealing-focus-on-my-mac) contains a python script that reports which app has the focus. Maybe some other app is having trouble with macOS Sonoma and grabbing the focus?

Don't think so. I was using several other apps, all of which I typically use, and they were all fine. Didn't do any drag & drop. Can't imagine why an app would want to communicate with IINA any other way (or even be able to). Although good to know about the script. I'll have it handy if it appears again.

**Comment by low-batt** [comment]
I'm mostly concerned about the mpv configuration, especially since it is generating errors.

The problem with the network theory is that we should be able to see that in the process sample. The process sample is missing IINA threads associated with the video view. It is what you might see if you take a sample of IINA without any windows open.

`youtube-dl` should not come into play unless a URL is opened.

I am familiar with the subtitles download code. I am rewriting part of it now. That code is only executed when the `Fine online subtitles from` menu item is clicked.

I'm definitely confused as to what could explain this.

I remember the `trustd` incident.

**Comment by svobs** [comment]
I see there is a `Search online subtitles automatically` checkbox, but I had that unselected.

If it turns out there's nothing making a network call where it shouldn't be, then I'm left scratching my head too. I don't understand what could cause a UI thread to hang for such a prolonged period, then come back.

I'm not really familiar with Apple's process samples, but I tried taking samples under a few different conditions. I was able to create a near-identical sample by opening a window and then pausing the video long enough for the DisplayLink to stop before sampling. 

I imagine system processes and/or calls are omitted from the process sample. I've noticed they have been when running profiling. So there may be pieces of the puzzle missing. But... ¯\_(ツ)_/¯
